api/v0/proto/retrievability/share_service.proto

```diff
@@ -0,0 +1,101 @@
+syntax = "proto3";
+
+package celestia.v0.retrievability.share;
+
+import "api/v0/proto/retrievability/options.proto";
+import "api/v0/proto/retrievability/types.proto";
+
+message GetRequest {
+  // The block height.
+  uint64 height = 1;
+  // The index in the ODS.
+  uint32 index = 2;
+  // Common options.
+  optional celestia.v0.retrievability.options.Options options = 3;
+}
+
+message GetResponse {
+  // The share.
+  optional celestia.v0.retrievability.types.Share share = 1;
+  // The proof of share inclusion.
+  optional celestia.v0.retrievability.types.DataRootProof proof = 2;
+}
+
+message GetODSRequest {
+  // The block height.
+  uint64 height = 1;
+  // Common options.
+  optional celestia.v0.retrievability.options.Options options = 2;
+}
+
+message GetODSResponse {
+  // The list of shares.
+  // TODO: does this include tail padding or no?
```

Comment:
Could be an optimisation if needed. I think we should return just square for sake of UX and improve padding overhead elsewhere 😆

Reply:
Clarified in ec5e64275c08392ffdfc1086b5cb4e40d1271e74